src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -98,11 +98,14 @@ private string GetRootPathWithGuid_NoLock()
                 Contract.ThrowIfNull(_workspace);
                 var tempPath = GetRootPathWithGuid_NoLock();
 
+                // We don't want to track telemetry for signatures only requests, only where we try to show source
```

Comment:
out of cursiosity, why?

Reply:
We are trying to understand which scenarios are more popular, and therefore could justify being worked on. If its a signature only request, then all of the scenarios are irrelevant anyway.

Also, in theory its possible to calculate the number of signature only requests, because it would be the number of peek definition or go to definition commands executed, minus the number of telemetry events we got from this (ie, number of non-signature only requests)

Reply:
>We are trying to understand which scenarios are more popular, and therefore could justify being worked on. If its a signature only request, then all of the scenarios are irrelevant anyway.

well... it tells us how popular they are *compared* to sig-only scenarios.  We can always ignore the sig-only number if we're just comparing the others :)

> Also, in theory its possible to calculate the number of signature only requests, because it would be the number of peek definition or go to definition commands executed, minus the number of telemetry events we got from this (ie, number of non-signature only requests)

sure... but i'm always worried about inferred stuff, rather than just a direct marker :)

Reply:
>  it tells us how popular they are compared to sig-only scenarios

The user doesn't control the type of request, so its not really popularity. We control it explicitly, and actually I am wrong about Peek Definition, because I changed that a little while ago. The only signatures only requests we have left are in LSP, so popularity of that scenario should be easy to track.

> i'm always worried about inferred stuff

You're also worried about telemetry that never gets used 😛